src/unix/fs.c

```diff
@@ -231,7 +231,29 @@ static ssize_t uv__fs_futime(uv_fs_t* req) {
 
 
 static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
-  return mkdtemp((char*) req->path) ? 0 : -1;
+  char *path = (char*) req->path;
+
+#if defined(__MVS__)
+  /* There is no mkdtemp. So instead use mktemp to generate a
```

Comment:
Interesting.  We can certainly do that after this.  Maybe what @jBarz is ok to land (with a TODO note) and then as a follow-up we factor out the Windows implementation?  This PR is already huge...